File: Demo/PayCardsNoAcquiringObjcDemo/ViewController.m
//
//  ViewController.m
//  PayCardsNoAcquiringObjcDemo
//
//  Created by Viktor Bespalov on 24/03/16.
//  Copyright © 2016 Pay.Cards. All rights reserved.
//

#import "ViewController.h"
#import <PayCardsNoAcquiring/PayCardsNoAcquiring.h>

@interface ViewController () <PCDCardScanViewControllerDelegete>

@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
@property (weak, nonatomic) IBOutlet UILabel *infoLabel;

@end

@implementation ViewController

#pragma mark - Lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBActions

- (IBAction)presentPayCards:(id)sender {
    
    PCDMerchant *merchant = [PCDMerchant new];
    merchant.ID = @100333; //Your id
    merchant.password = @"zI02NS3c"; // Your password
    merchant.clientId = @"777"; // Your client ID
    merchant.email = @"dmitriy.bagrov@walletone.com"; // Your email
    merchant.phone = @"79788014244"; // Your phone

    [self.activityIndicator startAnimating];
    [PCDCardScanViewController presentOnViewController:self merchant:merchant delegate:self verificationComplete:^(NSError *error) {
        [self.activityIndicator stopAnimating];
        
        if (error) {
            self.infoLabel.text = [NSString stringWithFormat:@"ERROR: %@", error.localizedDescription];
        }
    }];
}

#pragma mark - PCDCardScanViewControllerDelegete

- (void)cardScanViewController:(PCDCardScanViewController *)cardScanViewController didCloseWithError:(NSError *)error {
    self.infoLabel.text = [NSString stringWithFormat:@"ERROR: %@", error.localizedDescription];
    [cardScanViewController dismissWithCompletion:nil];
}

- (void)cardScanViewController:(PCDCardScanViewController *)cardScanViewController didFinishScanWithInfo:(NSDictionary *)cardInfo {
    self.infoLabel.text = [NSString stringWithFormat:@"CARD INFO: %@", cardInfo[PCDCardParams]];
    [cardScanViewController dismissWithCompletion:nil];
}

- (void)cardScanViewControllerUserDidClose:(PCDCardScanViewController *)cardScanViewController {
    self.infoLabel.text = @"RECOGNITION CANCELED";
    [cardScanViewController dismissWithCompletion:nil];
}

@end

File: Demo/PayCardsNoAcquiringSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  PayCardsNoAcquiringSwiftDemo
//
//  Created by Viktor Bespalov on 24/03/16.
//  Copyright © 2016 Pay.Cards. All rights reserved.
//

import UIKit
import PayCardsNoAcquiring

class ViewController: UIViewController, PCDCardScanViewControllerDelegete {

    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
    
    @IBOutlet weak var infoLabel: UILabel!
    
    // MARK: - Properties
    
    var merchant: PCDMerchant {
        let _merchant = PCDMerchant()
        _merchant.ID = 1 // Your id
        _merchant.password = "pass" // Your password
        _merchant.clientId = "456" // Your client ID
        _merchant.email = "ololo@qq.ee" // Your email
        _merchant.phone = "" // Your phone
        return _merchant
    }
    
    // MARK: - Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - IBActions
    
    
    @IBAction func presentPayCards(sender: AnyObject) {
        self.activityIndicator.startAnimating()
        PCDCardScanViewController.presentOnViewController(self, merchant: merchant, delegate: self) { (error) in
            self.activityIndicator.stopAnimating()
            
            if let error = error {
                self.infoLabel.text = "ERROR: \(error.localizedDescription)"
            }
        }
    }
    
    // MARK: - PCDCardScanViewControllerDelegete
    
    func cardScanViewController(cardScanViewController: PCDCardScanViewController!, didCloseWithError error: NSError!) {
        infoLabel.text = "ERROR: \(error?.localizedDescription)"
        cardScanViewController.dismissWithCompletion(nil)
    }
    
    func cardScanViewController(cardScanViewController: PCDCardScanViewController!, didFinishScanWithInfo cardInfo: [NSObject : AnyObject]!) {
        infoLabel.text = "CARD INFO: \(cardInfo[PCDCardParams])"
        cardScanViewController.dismissWithCompletion(nil)
    }
    
    func cardScanViewControllerUserDidClose(cardScanViewController: PCDCardScanViewController!) {
        infoLabel.text = "RECOGNITION CANCELED"
        cardScanViewController.dismissWithCompletion(nil)
    }
}

